src/main/scala/com/actian/spark_vector/colbuffer/ColumnBufferFactory.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016 Actian Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.actian.spark_vector.colbuffer
+
+import org.apache.spark.Logging
+
+import com.actian.spark_vector.colbuffer.integer._
+import com.actian.spark_vector.colbuffer.real._
+import com.actian.spark_vector.colbuffer.decimal._
+import com.actian.spark_vector.colbuffer.singles._
+import com.actian.spark_vector.colbuffer.string._
+import com.actian.spark_vector.colbuffer.time._
+import com.actian.spark_vector.colbuffer.timestamp._
+
+/** This is a `Factory` implementation of `ColumnBuffers`. */
+object ColumnBufferFactory extends Logging {
```

Comment:
For 2., I would go for something like this:

trait BufferBuilder {
def build: PartialFunction[(String, Int, String, Int, Int, Boolean, Int), ColumnBuffer[_]]
ColumnBuffer.register(build)
}

then in
object ColumnBuffer {
  private var builder: PartialFunction[(String, Int, String, Int, Int, Boolean, Int), ColumnBuffer[_]] = { case _ => throw new Exception("cannot find buffer for ... ") }
 def register(pt: PartialFunction[(String, Int, String, Int, Int, Boolean, Int), ColumnBuffer[_]]): Unit = { builder = pt orElse builder }
 def apply(name, Index, .... ) = builder(name, Index, .... )
}

and finally
object DecimalIntBuffer extends BufferBuilder { 
def build = { case (name, Index, ...) => new DecimalIntbuffer }
}